### ISSUE TITLE
NUMBA vs NUMPY performance and diag_ fields missing.

### ISSUE THREAD

**Issue by apiszcz** [soft-matter/trackpy#226]
1. not sure what happened in my earlier test where numba was faster than numpy, now I'm seeing numpy faster on small problems.
2. I tried to enable diagnostics, however I'm not seeing any of the diag_ being created, I'll probably step into the code to see what is going on.

**Comment by nkeim** [comment]
1. This is a known "issue": the pure-Python code (no numpy) is actually *slightly* faster than the numba code for trivial subnets. However, for large subnets, the numba code is *orders of magnitude* faster. That is why we have made numba the default. But you may get a boost from using one of the pure-Python linkers instead. We're aware of this and may eventually allow trackpy to automatically choose the implementation that is likely to be fastest. We would just need to figure out how to make that decision efficiently.

2. When you get the chance, please let us know how to reproduce the problem. And if you realize that the problem is due to something confusing in the documentation, we'd like to know about that too!

**Comment by apiszcz** [comment]
Has anyone considered a quarterly or less frequent teleconference to discuss topics about new capabilities, performance, use cases etc,?

**Comment by danielballan** [comment]
Sure, that sounds worthwhile. Some of us used to organize a meetup at the APS march meeting, though it didn't happen this year, and of course that doesn't reach everyone who might be interested.

Currently, we are prioritizing polishing exiting features over new enhancements, as much as been added in the last several months and we'd like to get v0.3 released. But it's never too early to discuss ideas, and it would be fun to get our little community together. Maybe a Google Hangouts sometime in the coming weeks?

**Comment by scabi** [comment]
I am testing numpy and numba for "locate" method. On my system numba is slower 2-6 times! I am using a Mac Anaconda Python 3.4 with all the latest package updates. My current workload is 5M pixel image sequences with couple hundred about 40x40 pixel size features each.
Is this normal, is it something we can troubleshoot?

**Comment by nkeim** [comment]
Yes, we're always interested in unusual cases like this. Can you tell us what version of `trackpy` you're using? Assuming you used a method like `pip`, `conda`, `setup.py`, etc. to install it, this is just the contents of `trackpy.__version__`. The reason I ask is that a similar problem is addressed in the development branch, via #217.

The above discussion was for the subnet solver (during the linking step), so if this is a new problem, we should open a new issue.

**Comment by scabi** [comment]
Soft-matter channel did not install (had some conflicts with conda python3.4) so I installed it from danielballan's channel where version is listed v0.2.4.post93. trackpy.__version__ gives 'unknown' though. I can force an install from the official channel if needed.

I wasn't sure if something common causes these performance issues. Feel free to open a new ticket if you think that is more appropriate. I can provide some sample images if someone wants to investigate further.

**Comment by apiszcz** [comment]
I had similar issues with the install and used the .zip file and have no
issues in general. Though __version__ is not set. This was raised earlier
and may be fixed by now. I'm using the latest dev release..

**Comment by danielballan** [comment]
I have gotten a little better at packaging since the last time I dug into
trackpy's build process. I think I can address most of these problems.

**Comment by apiszcz** [comment]
Have there been any estimates on the when we might see next release
candidate or full release?

**Comment by danielballan** [comment]
Not recently. There are several giant PRs to review and tests. Speaking just for myself, I think we can put out a release candidate in June or early July.

**Comment by danielballan** [comment]
It's hard to diagnose __version__ issues, but I am hopeful that #262 may fix that problem. If you are able to run your code on the current master branch, I'm interested in revisiting your original question about performance.
